src/fieldValidation.ts

```diff
@@ -3,9 +3,21 @@ import addFormats from 'ajv-formats';
 import { phone } from 'phone';
 import { BaseFieldDataType } from './types';
 
-const ajv = new Ajv();
+const ajv = new Ajv({ coerceTypes: true });
 addFormats(ajv, ['email', 'uri']);
 
+function isValidNumber(num: string): boolean {
```

Comment:
A few things which apply to ALL of these functions

1.  We use arrow functions pretty much 100% of the time; they're better than functions (@jasonaowen and I had a super long exploration of this on another project and... I cannot find the thread at all).  The big benefit is that it protects you from weird scoping issues and dangerous access of `this`.

2.  We shouldn't name the variable `num` since the point of this is that we don't know if it's a num or not!  I'd name it something generic like `value` instead.

3. You could probably get away with typing the input as `unknown` (the idea is you're checking if a generic value is a number, so, if it's literally a number then that's fine too!)

4. Realizing you can (should?) remove `valid` since it's somewhat redundant.  For this specific case I would strongly consider `isNumericString` instead of `isNumber` since `isNumber` is going to be confusing.  Similarly, `isBooleanString` instead of `isBoolean`.

5. (This overrides 1-3... but I felt those were worth saying in general for edification!) `ajv` has a nifty function that will take a schema and return a validator.  It's `ajv.compile`.  You can see some examples of this in our types (for instance: https://github.com/PhilanthropyDataCommons/service/blob/e109501b2ce4cff25f25b7ba90050108135ac87d/src/types/Organization.ts#L27)

This doesn't apply here, but one cool thing about compile is that it also serves as a type guard (but that doesn't quite work).  It also populates itself with specific errors about why validation failed, which could be handy.

Using this you'd be able to turn your manually written functions into:

```
const isNumericString = ajv.compile({
  type: 'number',
});
```

Reply:
Type-guards! These rock, thank you for the suggestion, and I appreciate the notes for if we weren't using ajv, they helped with making a wrapping function for the phoneNumber validation.